docs/HowToConnectExternally.md

```diff
@@ -0,0 +1,298 @@
+
+## Transaction Engine Groups
+
+A TE group consists of TEs with the same configuration which is part of the same database most often used to serve specific SQL workload.
+NuoDB Helm Charts 3.4.0+ supports the deployment of one or more TE groups per database.
+
+Multiple Helm releases of the `database` chart for the same NuoDB database can be installed in the same Kubernetes namespace where only one of them is _primary_.
+One or more _secondary_ Helm releases are used to deploy additional TE groups for the same database with different configuration options.
+This allows flexible configuration of each TE group including but not limited to the number of TEs in a group, their resource requirements, process labels, and scheduling rules.
+SQL clients can be configured to target each TE group separately using NuoDB Admin load balancer rules.
+Specifying the type of the Helm `database` release is controlled by the  `database.primaryRelease` option (`true` by default).
+
+## External Access for TE Groups
+
+External access for the NuoDB database is _not_ enabled by default.
+To enable external access to the NuoDB domain and database, set the `admin.externalAccess.enabled=true` and `database.te.externalAccess.enabled=true` options.
+Configuring external database access via the NuoDB Admin with TE groups is supported with NuoDB Helm Charts 3.4.0+ and NuoDB 4.2.4+.
+In earlier versions, it is only possible to connect to the provisioned TEs directly and bypass the NuoDB Admin.
+
+A Kubernetes service of type `LoadBalancer` or `NodePort` is created per TE group.
+This should allow the external SQL clients to connect to the TEs backing the service by uniquely targeting each TE group.
+Most of the cloud vendors provide Kubernetes Load Balancer controllers that support different service annotations used to control the properties and configuration of the provisioned cloud load balancer.
+The Kubernetes cluster should be properly configured so that the external network (Layer4) cloud load balancer is provisioned automatically.
+For more information on additional cluster configuration, see [Cloud Provider Specifics](#cloud-provider-specifics).
+
+> **NOTE**: When external access is enabled, the NuoDB Helm Charts will create Internet-facing load balancers by default.  Be sure to understand the difference between _internet facing_ load balancers (allowing connectivity external to the cloud and Kubernetes cluster) and  _internal_ load balancers (allowing connectivity external to the Kubernetes cluster but within the cloud virtual network).
```

Comment:
What mechanism are you referring to when you say "internal load balancers"? "external to the Kubernetes cluster but within the cloud virtual network" sounds like it would be the same mechanism, e.g. `type=LoadBalancer`, but the DNS entry that is created is only resolvable within a private network.

Reply:
So it looks like the mechanism here is using `internalIP`, which seems to be available in AWS by specifying certain annotations. Is it available in the other cloud providers, and if not should we make a note about it?

Reply:
This note here is specific to the load balancers provisioned in the cloud. The `database.te.externalAccess.internalIP` and `admin.externalAccess.internalIP` works in all supported cloud providers.